src/serial/MSXRS232.cc

```diff
@@ -241,10 +242,13 @@ byte MSXRS232::readStatus(EmuTime::param time)
 	auto& dev = getPluggedRS232Dev();
 
 	// Mask out (active low) bits
-	if (dev.getDCD(time)) {
+	auto dcd = dev.getDCD(time);
+	auto ri = dev.getRI(time);
+
+	if (!dcd ? inputsPullup : *dcd) {
```

Comment:
This code is correct.

It could be simplified a bit by using some of the convenience functions offered by `std::optional`, in particular:
    https://en.cppreference.com/w/cpp/utility/optional/value_or

`value_or(<default>)` returns the value contained in the optional, or a default value if the optional was empty.

And with that the code simplifies to:

    if (dev.getDCD(time).value_or(inputsPullup)) result &= ~0x01;
    ...
    if (hasRIPin && dev.getRI(time).value_or(inputsPullup)) result &= ~0x02;

And similar below:

    return rs232.getPluggedRS232Dev().getDSR(time).value_or(rs232.inputsPullup);
    ....
    return rs232.getPluggedRS232Dev().getCTS(time).value_or(rs232.inputsPullup);

Reply:
Ah yes, that's more readable, I'll make the change.